File: julia/client/json_client.jl
#
# JSON client with Julia
#
# author: Atsushi Sakai
#

using HTTP
using JSON

function get_config()
    ip = "localhost" #default
    if haskey(ENV,"SERVER_IP")
        ip = ENV["SERVER_IP"] 
    end

	port = "8000" #default
    if haskey(ENV,"SERVER_PORT")
        port = ENV["SERVER_PORT"]
    end

    println("request to ip:", ip)
    println("request to port:", port)

    return ip, port
end

function main()
	req = Dict("Name" => "julia json request")
	println("request:")
	println(req)

    ip, port = get_config()

    url ="http://"*ip*":"*port
  	r = HTTP.request("POST", url,
               ["Content-Type" => "application/json"],
               JSON.json(req))

    println("response:")
  	println(JSON.parse(String(r.body)))

    println(PROGRAM_FILE," Done!!")
end

main()

File: python/client/json_client.py
"""

JSON Client with python

author: Atsushi Sakai (@Atsushi_twi)

"""

import requests
import json
import os


def get_config():
    ip = os.environ.get('SERVER_IP', 'localhost')
    port = os.environ.get('SERVER_PORT', '8000')
    return ip, port


def main():

    ip, port = get_config()

    url = "http://" + ip + ":" + port
    print("url:", url)

    headers = {"content-type": "application/json"}
    obj = {"Type": "python json client", 123: 123}
    json_data = json.dumps(obj)
    print("request:")
    print(json_data)
    response = requests.post(url, headers=headers, data=json_data)

    res = response.json()
    print("response:")
    print(res)


if __name__ == '__main__':
    main()
